Mage.Sets/src/mage/cards/s/SanctuaryBlade.java

```diff
@@ -25,13 +29,15 @@ public SanctuaryBlade(UUID ownerId, CardSetInfo setInfo) {
         this.subtype.add(SubType.EQUIPMENT);
 
         // As Sanctuary Blade becomes attached to a creature, choose a color.
-        GainProtectionFromColorAttachedEffect protectionEffect = new GainProtectionFromColorAttachedEffect(Duration.WhileOnBattlefield);
-        this.addAbility(new AttachedToCreatureTriggeredAbility(protectionEffect, false));
+        this.addAbility(new AttachedToCreatureSourceTriggeredAbility(new ChooseColorEffect(Outcome.Benefit), false));
 
-        // Equipped creature gets +2/+0 and has protection from the last chosen color.
```

Comment:
That was fine comment -- full description from rules -- don't split it.

1. So you must have one static ability with two effects from ability (use `ability.AddEffect`), not two different abilities;
2. Also be careful with effects names -- you can add delimeter between effect texts as `.concatBy("end")` or/and setup custom effect names. Search `showCardInfo()` in `VerifyCardDataTest.java` -- it's useful test to check your card rules runtime without db clean.

Reply:
ah gotcha, makes sense, fixed